sycl/test-e2e/ESIMD/local_accessor_copy_to_from.cpp

```diff
@@ -0,0 +1,135 @@
+//==-- local_accessor_copy_to_from.cpp  - DPC++ ESIMD on-device test -------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+//
+// TODO: Enable the test when GPU driver is ready/fixed.
+// XFAIL: opencl || windows || gpu-intel-pvc || gpu-intel-gen12
+// TODO: add support for local_accessors to esimd_emulator.
+// UNSUPPORTED: esimd_emulator
+// The test checks functionality of the gather/scatter local
+// accessor-based ESIMD intrinsics.
+
+#include "esimd_test_utils.hpp"
+
+#include <iostream>
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+using namespace sycl;
+
+template <typename T, unsigned VL> bool test(queue q) {
+  constexpr size_t size = VL;
+
+  std::cout << "Testing T=" << typeid(T).name() << " VL=" << VL << "...\n";
+
+  // The test is going to use size elements of T type.
+  auto Dev = q.get_device();
+  auto DeviceSLMSize = Dev.get_info<sycl::info::device::local_mem_size>();
+  if (DeviceSLMSize < size) {
+    // Report an error - the test needs a fix.
+    std::cerr << "Error: Test needs more SLM memory than device has!"
+              << std::endl;
+    return false;
+  }
+
+  T *A = new T[size];
+
+  for (unsigned i = 0; i < size; ++i) {
+    A[i] = static_cast<T>(i);
+  }
+
+  try {
+    buffer<T, 1> buf(A, range<1>(size));
+    nd_range<1> glob_range{range<1>{1}, range<1>{1}};
+
+    q.submit([&](handler &cgh) {
+       auto acc = buf.template get_access<access::mode::read_write>(cgh);
+       auto LocalAcc = local_accessor<T, 1>(size, cgh);
+       cgh.parallel_for(glob_range, [=](id<1> i) SYCL_ESIMD_KERNEL {
+         using namespace sycl::ext::intel::esimd;
+         simd<T, VL> valsIn;
+         simd<T, VL> valsOut;
+         valsIn.copy_from(acc, 0);
+         valsIn += 1;
+         valsIn.copy_to(LocalAcc, 0);
+         valsOut.copy_from(LocalAcc, 0);
+         valsOut.copy_to(acc, 0);
```

Comment:
This variant leaves lots of space for double error in API implementation (1 for write and 1-same for read), which would result into pass.

There should be several ways how to make checks stricter, just for example:
- write 1 element to offset=0;
- write 2 elements to offset=sizeof(T);
- ...
- write VL elements to end of allocated SLM
- 
- - barrier

- read VL elements (copy_from) VL elems from offset=0, 
- read VL/2 elems from offset=VL*sizeof(T), etc.
...
- read 1 element from end of allocated SLM.
It would do some cross checks for read-writes.